src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,159 @@
+            throw new DukeException(EVENT_DESCRIPTION_ERROR);
+        }
+        if (!line.contains("/at")) {
+            throw new DukeException(EVENT_EMPTY_ERROR);
+        }
+        String[] input = line.substring(EVENT_DESCRIPTION_START).split("/");
+        tasks[currCount] = new Event(input[0], input[1].substring(EVENT_DEADLINE_TIME));
+    }
+
+    public static void addDeadline(Task[] tasks, int currCount, String line) throws DukeException {
+        if (line.length() < DEADLINE_MIN_LENGTH) {
+            throw new DukeException(DEADLINE_DESCRIPTION_ERROR);
+        }
+        if (!line.contains("/by")) {
+            throw new DukeException(DEADLINE_EMPTY_ERROR);
+        }
+        String[] input = line.substring(DEADLINE_DESCRIPTION_START).split("/");
+        // first elem: description, second elem: deadline
+        tasks[currCount] = new Deadline(input[0], input[1].substring(EVENT_DEADLINE_TIME));
+    }
+
+    public static void addTodo(Task[] tasks, int currCount, String line) throws DukeException {
+        if (line.length() < TODO_MIN_LENGTH) {
+            throw new DukeException(TODO_DESCRIPTION_ERROR);
+        }
+        // take the description part of the input string
+        tasks[currCount] = new Todo(line.substring(TODO_DESCRIPTION_START));
+    }
+
+    public static void printGoodBye() {
+        printHorizontalLine();
+        System.out.println(" Bye. Hope to see you again soon!");
+        printHorizontalLine();
+    }
+
+    public static void main(String[] args) {
+        printWelcomeMessage();
+        Task[] tasks = new Task[MAX_TASKS]; // fixed size array for now, each contains a Task element
+        int currCount = 0;
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+        while (!line.equals("bye")) {
+            try {
+                if (line.contains("done")) { // mark task as done
+                    markAsDone(tasks, currCount, line);
+                } else if (line.contains("todo")) {
+                    addTodo(tasks, currCount, line);
+                    currCount = completeAddTask(tasks, currCount);
+                } else if (line.contains("deadline")) {
+                    addDeadline(tasks, currCount, line);
+                    currCount = completeAddTask(tasks, currCount);
+                } else if (line.contains("event")) {
+                    addEvent(tasks, currCount, line);
+                    currCount = completeAddTask(tasks, currCount);
+                } else if (line.equals("list")){ // print the list
+                    printList(tasks, currCount);
+                }
+                else {
+                    throw new DukeException(COMMAND_ERROR);
+                }
+            } catch (DukeException e) {
+                printHorizontalLine();
+                System.out.println(e.getMessage());
+                printHorizontalLine();
+            }
+            line = in.nextLine();
+        }
+        printGoodBye();
+    }
```

Comment:
You should minimize the size of the main method by splitting the different functionalities into their own methods. By reducing the size of the method and ensuring each method does one functionality it is supposed to do, your code is easier to maintain, scale and debug.